sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1787,6 +1787,13 @@ object SQLConf {
     val SQL_STANDARD, ISO_8601, MULTI_UNITS = Value
   }
 
+  val OPTIONAL_INTERVAL =
+    buildConf("spark.sql.optionalInterval")
```

Comment:
let's put it under the `spark.sql.parser` namespace

Reply:
ok